apps/opencs/view/render/terraintexturemode.hpp

```diff
@@ -0,0 +1,137 @@
+#ifndef CSV_RENDER_TERRAINTEXTUREMODE_H
+#define CSV_RENDER_TERRAINTEXTUREMODE_H
+
+#include "editmode.hpp"
+
+#include <string>
+
+#include <QWidget>
+#include <QLabel>
+#include <QSpinBox>
+#include <QGroupBox>
+#include <QSlider>
+#include <QIcon>
+#include <QFrame>
+#include <QEvent>
+#include <QHBoxLayout>
+#include <QPushButton>
+
+#include "../../model/world/data.hpp"
+#include "../../model/world/land.hpp"
+
+namespace CSVWidget
+{
+    class SceneToolMode;
+}
+
+namespace CSVRender
+{
+    class BrushSizeControls : public QGroupBox
+    {
+        Q_OBJECT
+
+        public:
+            BrushSizeControls(const QString &title, QWidget *parent);
+            QSlider *brushSizeSlider;
+
+        private:
+            QSpinBox *brushSizeSpinBox;
+            QHBoxLayout *layoutSliderSize;
+    };
+
+    class TextureBrushButton : public QPushButton
+    {
+        Q_OBJECT
+
+        public:
+            TextureBrushButton (const QIcon& icon, const QString& tooltip = "",
+                QWidget *parent = 0);
+    };
+
+    class TextureBrushWindow : public QFrame
+    {
+        Q_OBJECT
+
+        public:
+            TextureBrushWindow(WorldspaceWidget *worldspaceWidget, QWidget *parent = 0);
+            void configureButtonInitialSettings(TextureBrushButton *button);
+
+            TextureBrushButton *buttonPoint = new TextureBrushButton(QIcon (QPixmap (":scenetoolbar/brush-point")), "", this);
+            TextureBrushButton *buttonSquare = new TextureBrushButton(QIcon (QPixmap (":scenetoolbar/brush-square")), "", this);
+            TextureBrushButton *buttonCircle = new TextureBrushButton(QIcon (QPixmap (":scenetoolbar/brush-circle")), "", this);
+            TextureBrushButton *buttonCustom = new TextureBrushButton(QIcon (QPixmap (":scenetoolbar/brush-custom")), "", this);
+
+        private:
+            QLabel *selectedBrush;
+            QGroupBox *horizontalGroupBox;
+            int mButtonSize;
+            int mBrushSize = 0;
```

Comment:
Its not a good idea to initialise variables here if you also have a constructor, because that separates the initialisation into two different parts of the code (two different files actually) which makes it harder to verify that everything got initialised correctly.